p2p/swarm.go

```diff
@@ -218,12 +218,17 @@ func (s *swarm) SendMessage(peerPubKey string, protocol string, payload []byte)
 		Payload:  payload,
 	}
 
-	err = message.SignMessage(s.lNode.PrivateKey(), protomessage)
+	data, err := proto.Marshal(protomessage)
 	if err != nil {
-		return err
+		e := fmt.Errorf("invalid msg format %v", err)
+		return e
 	}
 
-	data, err := proto.Marshal(protomessage)
+	sign, _ := session.Sign(data)
+
+	protomessage.Metadata.AuthorSign = sign
```

Comment:
Should AuthorSign be an HMAC signature ? if this message will be passed (gossip) the receivers might not be able to verify it.

Reply:
Maybe we should add another sign, that was the original idea of `AuthorSign`.

Reply:
So as discussed we should have both